### DISCUSSION TITLE
What kind of engine is used?

### DISCUSSION THREAD

**Discussion by VuzzyM** [KiraLT/torrent-stream-server#216] [Q&A] [answered]
Libtorrent?

do you support these protocols?

Protocol obfuscation
DHT
uTP
PEX
Magnet links
IP Blocklists
Some IPv6
HTTP and UDP tracker clients

BEPs:

3: Basic BitTorrent protocol
5: DHT
6: Fast Extension (have all/none only)
7: IPv6 Tracker Extension
9: ut_metadata
10: Extension protocol
11: PEX
12: Multitracker metadata extension
15: UDP Tracker Protocol
20: Peer ID convention ("-GTnnnn-")
23: Tracker Returns Compact Peer Lists
27: Private torrents
29: uTorrent transport protocol
41: UDP Tracker Protocol Extensions
42: DHT Security extension
43: Read-only DHT Nodes

BEPs:

14: Local Peers Discovery
19: WebSeeds

**Comment by omgbox** [comment]
I believe it is currently using github.com/webtorrent/webtorrent only UDP TCP  , but would be much nicer to use https://github.com/webtorrent/webtorrent-hybrid , as it also connects to WebRTC users. i am not sure if the webtorrent right now uses WebRTC.

**Comment by KiraLT** [comment, marked as answer]
We use [Webtorrent](https://github.com/webtorrent/webtorrent) for streaming support, you can find all information on their Github. Not all functionality is implemented, so some features can be missing. Could you tell more specifically what you are looking for or maybe you are missing some functionality?

**Reply by lu4t** [reply]
is the server connecting to the bittorrent network? webtorrent? or hybrid?. It is unclear to me when I read the answers.
When I try to play an infohash, which is working when I use other bittorrent clients, I am not able to play it with torrent-stream-server... and I was wondering if the reason is that only webtorrent peers are searched.

Thanks for this great code, I am a big supporter and active user of it.

**Reply by KiraLT** [reply]
According to the docs:

> In node.js, the `webtorrent` package only connects to normal TCP/UDP peers, not WebRTC peers. If you want to connect to all types of peers, including WebRTC peers, from node.js, you need to use this package (`webtorrent-hybrid`).

So it is probably a good idea to switch to [webtorrent/webtorrent-hybrid](https://github.com/webtorrent/webtorrent-hybrid), I added it to the plans: #303
